pkg/election/election.go

```diff
@@ -385,13 +391,42 @@ forLoop:
 	return session, err
 }
 
+// ClearSessionIfNeeded will clear session when deleted master quited abnormally
+// returns (triggered deleting session, error)
+func (e *Election) ClearSessionIfNeeded(ctx context.Context, id string) (bool, error) {
+	resp, err := e.cli.Get(ctx, e.key, clientv3.WithPrefix())
+	if err != nil {
+		return false, err
+	}
+	deleteKey := ""
+	for _, kv := range resp.Kvs {
+		leaderInfo, err2 := getCampaignerInfo(kv.Value)
+		if err2 != nil {
+			return false, err2
+		}
+		if leaderInfo.ID == id {
+			deleteKey = string(kv.Key)
+			break
+		}
+	}
+	if deleteKey == "" {
+		// no campaign info left in etcd, no need to trigger re-campaign
+		return false, nil
+	}
+	delResp, err := e.cli.Delete(ctx, deleteKey)
```

Comment:
what error message will get for `elec.Campaign`(L244) when deleting the key?

Reply:
It will get no error but a delete event